docs/dev/libs/state_processor_api.md

```diff
@@ -345,6 +345,44 @@ Along with reading registered state values, each key has access to a `Context` w
 
 {% panel **Note:** When using a `KeyedStateReaderFunction`, all state descriptors must be registered eagerly inside of open. Any attempt to call a `RuntimeContext#get*State` will result in a `RuntimeException`. %}
 
+### Window State
+
+The state processor api supports reading state from a [window operator]({{ site.baseurl }}/dev/stream/operators/windows.html).
+This includes both time based windows along with other types, pre-aggregation, non-aggregated, and windows with evictors.
+
+<div class="codetabs" markdown="1">
+<div data-lang="java" markdown="1">
+{% highlight java %}
+
+ExecutionEnvironment batchEnv = ExecutionEnvironment.getExecutionEnvironment();
+ExistingSavepoint savepoint = Savepoint.load(batchEnv, "hdfs://checkpoint-dir", new MemoryStateBackend());
+
+int count = savepoint
+    // The timeWindow method supports reading from any type of time based window, including but not limited to
+    // Tumbling, Sliding, and Session windows for both event time and processing time.
+    .timeWindow()
+    .reduce(uid, new ReduceSum(), Types.INT, Types.INT)
+    .count();
```

Comment:
I don't understand this example. Where does the count() method come from, and what is being counted? 

If I wanted to inspect or modify the window contents, how would I do that? Can I write window state, or only read it?

Reply:
`count` is a dataset feature. Will remove.